streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -481,10 +479,24 @@ private static int getCrossRackTrafficCost(final Set<TaskTopicPartition> topicPa
      */
     private static boolean canPerformRackAwareOptimization(final ApplicationState applicationState,
                                                            final AssignedTask.Type taskType) {
-        final String rackAwareAssignmentStrategy = applicationState.assignmentConfigs().rackAwareAssignmentStrategy();
+        final AssignmentConfigs assignmentConfigs = applicationState.assignmentConfigs();
+        final String rackAwareAssignmentStrategy = assignmentConfigs.rackAwareAssignmentStrategy();
         if (StreamsConfig.RACK_AWARE_ASSIGNMENT_STRATEGY_NONE.equals(rackAwareAssignmentStrategy)) {
+            LOG.warn("Rack aware task assignment optimization disabled: rack aware strategy was set to {}",
+                rackAwareAssignmentStrategy);
+            return false;
+        }
+
+        if (!assignmentConfigs.rackAwareTrafficCost().isPresent()) {
+            LOG.warn("Rack aware task assignment optimization unavailable: the traffic cost configuration was not set.");
```

Comment:
We should log the exact config name since otherwise people won't necessarily know what this is referring to (especially since they already forgot to set this config). 

```suggestion
            LOG.warn("Rack aware task assignment optimization unavailable: must configure {}", StreamsConfig.RACK_AWARE_ASSIGNMENT_TRAFFIC_COST_CONFIG);
```